[Ingest pipelines] Improve copy of Reroute processor

ElenaStoeva opened the pull request:
Closes https://github.com/elastic/kibana/issues/162848

## Summary

This PR improves the ui copy of the Reroute processor form and some common processor components.

List of changes that have been made:

- Title of Processor form
- Processor field placeholder
- Destination field helper text
- Dataset field placeholder and helper text
- Namespace field placeholder and helper text
- Condition field helper text
- Tag field helper text
- Ignore failure field label
- Submit button label

<details>
<summary>Screenshots</summary>
<img width="535" alt="Screenshot 2023-08-14 at 18 41 57" src="https://github.com/elastic/kibana/assets/59341489/473bdef1-7f00-4002-8ae3-0bf104bfdebd">


The `index names` link in the Dataset and Namespace helper text leads to https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-create-index.html#indices-create-api-path-params

<img width="715" alt="Screenshot 2023-08-02 at 17 03 26" src="https://github.com/elastic/kibana/assets/59341489/1b026562-23ff-443f-8d8a-7dfff2455ea5">

</details>


### Checklist

- [X] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [X] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [X] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))

elasticmachine commented:
Pinging @elastic/platform-deployment-management (Team:Deployment Management)

alisonelizabeth reviewed (approved):
Code changes LGTM. Verified locally. I'll defer to the docs team on the copy itself.

alisonelizabeth commented:
@elasticmachine merge upstream

gchaps commented:
I did a peer review on the copy, and here are some comments:

- One thing I didn’t realize is that the changes for **Condition** and **Ignore missing** affect other processors.  Does the copy work for the other processors as well?

- Are **dataset** and **namespace** familiar terms?  If not, can we put tooltips on those labels?

- "A static value for the target".  If that's the target index, how about "A value for the target index." Can static be removed?

- "An identifier for the processor" sounds better than "label".

ElenaStoeva commented:
Thanks for the review @gchaps! I updated the screenshot in the PR description with the new changes.

> * One thing I didn’t realize is that the changes for **Condition** and **Ignore missing** affect other processors.  Does the copy work for the other processors as well?

Yes, these fields have the same context for all processors so their copy change should be okay.

-------

> * Are **dataset** and **namespace** familiar terms?  If not, can we put tooltips on those labels?

I'm not sure if they are familiar terms so I added a link in the helper text to the [doc](https://www.elastic.co/guide/en/fleet/master/data-streams.html#data-streams-naming-scheme) that explains these terms in detail (I don't think there is an easy way to add a tooltip to the label).

------

> * "A static value for the target".  If that's the target index, how about "A value for the target index." Can static be removed?

Yes, I think it would make sense to remove "static".

-----

> * "An identifier for the processor" sounds better than "label".

👍 

----

> It looks like you can route a document to either a target index, in which you need to set a destination, or to a data stream, where you need to enter a dataset and namespace. Can the UI include a switch where users select whether they are using an index or data stream? Then the UI would only show the fields where users need to provide input? That would eliminate the need for copy such as "Not available when dataset or namespace is set" and many of the "Optional" labels.

I think this is a great idea!
Since this would require more involved work, I will create a separate issue for this improvement and we can revisit it when the team has bandwidth.

ElenaStoeva commented:
@elasticmachine merge upstream

gchaps reviewed (approved):
LGTM

ElenaStoeva commented:
@elasticmachine merge upstream
